Binary/Loader/FileSystemLoader.php

```diff
@@ -38,30 +38,41 @@ class FileSystemLoader implements LoaderInterface
     /**
      * @param MimeTypeGuesserInterface  $mimeGuesser
      * @param ExtensionGuesserInterface $extensionGuesser
-     * @param string[]                  $dataRoots
-     * @param LocatorInterface          $locator
+     * @param LocatorInterface|string[] $locatorOrDataRoots
      */
     public function __construct(
         MimeTypeGuesserInterface $mimeGuesser,
         ExtensionGuesserInterface $extensionGuesser,
-        $dataRoots
-        /* LocatorInterface $locator */
+        $locatorOrDataRoots
     ) {
         $this->mimeTypeGuesser = $mimeGuesser;
         $this->extensionGuesser = $extensionGuesser;
 
-        if (count($dataRoots) === 0) {
-            throw new InvalidArgumentException('One or more data root paths must be specified.');
-        }
+        if (is_array($locatorOrDataRoots)) { // BC
+            if (func_num_args() === 4 && func_get_arg(3) instanceof LocatorInterface) {
```

Comment:
Sorry, I didn't even realize you'd removed `setOptions` from the `Loader`; that *needs* to stay with a deprecation notice. Then, in the `Locator` constructor, you should pass the array to `setOptions`, as before. As far as the other changes, of defaulting to 2.0 usage, that's fine, as long as we support the old use cases.